templates/ui.html

```diff
@@ -871,6 +871,19 @@
           primary: '#5a3cff',
         },
       }
+      function uuidv4() {
+        return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+          (c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+        );
+      }
+
+      function randomId() {
+          if (typeof crypto.randomUUID == "function") {
+              return crypto.randomUUID();
+          } else {
+              return uuidv4();
+          }
+      }
```

Comment:
Put this in different js file in `static/js`, and you need to include crypto js lib to use crypto functions

Reply:
moved to static/js/utils.js